UI/test.js

```diff
@@ -1,50 +1,142 @@
-            author: 'Ivan',
-            photoLink: 'fhcgjhbjj;lk'
-        })
-    }
-    catch (error) {
-        console.warn(error)
-    }
-
-    try {
-        console.log('Validating post with wrong date')
-        Oazis.validatePost({
-            id: '4',
-            description: 'dog',
-            createdAt: [1, 2, 3],
-            author: 'Ivan',
-            photoLink: 'fgjklj;k'
-        })
-    }
-    catch (error) {
-        console.warn(error)
-    }
-
-    console.log('Adding new post')
-    console.log(Oazis.addPost({
-        id: '11',
-        description: 'doggy',
-        createdAt: new Date('2018-03-03'),
-        author: 'Anna',
-        photoLink: 'https://placeimg.com/300/300/animals'
-    }))
-    console.log(Oazis.getPosts())
-
-    console.log('Removing post with id = 1')
-    Oazis.removePost('1')
-    console.log(Oazis.getPosts())
+window.client = (function () {
+    let user = 'anniepank'
+    let isLoggedIn = false
+    let filters = {}
+    let posts = []
+    let postList = document.getElementsByClassName('post-list')[0]
+
+    function createPostElement(postConfig) {
+        if (postConfig.deleted) return;
+
+        let postElement = document.getElementById('templatePost').cloneNode(true)
+
+        postElement.id = null
+
+        postElement.querySelector('.nickname').innerText = postConfig['author']
+        postElement.querySelector('.date').innerText = postConfig['createdAt'].toDateString()
+
+        postElement.querySelector('.post-description').innerText = postConfig['description']
+        postElement.querySelector('.post-hashtags').innerText = postConfig['hashtags']
+
+        if ((user === null) || (postConfig.author !== user)) {
+            postElement.querySelector('.to-the-right').querySelector('.fa-trash').style.display = 'none'
+            postElement.querySelector('.to-the-right').querySelector('.fa-pencil-alt').style.display = 'none'
+        }
+
+        if (postConfig.likes)
+        postElement.querySelector('#likes').innerText = postConfig.likes
+
+        postList.appendChild(postElement)
+
+        return postElement
+    }
+
+    function addPost(post) {
+        Oazis.addPost(post)
+        loadMore(1)
+        reload()
+    }
+
+    function reload() {
+        let n = posts.length
+        posts = []
+        loadMore(n)
+        update()
+    }
+
+
+    function update() {
+        while (postList.firstChild) {
+            postList.removeChild(postList.firstChild);
+        }
+
+        for (let post of posts) {
+            createPostElement(post)
+        }
+
+    }
+
+    function loadMore(numberOfPosts, filters) {
+        let additionalPosts = Oazis.getPosts(posts.length, numberOfPosts, filters);
+        posts = posts.concat(additionalPosts)
+        update()
+    }
+
+    function filterPosts(filterConfig) {
+        let limit = posts.length
+        posts = Oazis.getPosts(0, limit, filterConfig)
+        update()
+    }
+
+    function removePost(post) {
```

Comment:
Sorry, do you really need to pass the whole post object as an argument? It it possible to use just id?